spec/std/kernel_spec.cr

```diff
@@ -220,3 +220,22 @@ describe "at_exit" do
                            OUTPUT
   end
 end
+
+describe "seg fault" do
+  it "detects stack overflow" do
```

Comment:
How about a spec which stack overflows the main stack, not just a fiber stack?

How long does this spec run for (time)?

Reply:
Each spec in `kernel_spec.cr` costs about 3 seconds apiece to run for a total of 36 seconds on my laptop.  The one I added is no different.  So I was leery about adding too many of these. 

In my actual testing on the various platforms I used the following:

* Random pointer deref does not trigger stack overflow
* Trivial infinite recursion on main stack triggers stack overflow
* Infinite recursion on main stack with modest stack allocations triggers stack overflow
* Trivial infinite recursion on fiber stack triggers stack overflow (this became the official spec)
* Infinite recursion on fiber with modest stack allocation triggers stack overflow

All of these take about the same amount of time, except on one of the BSDs (freebsd, I think) the default stack size is gargantuan and `ulimit` didn't seem to work to restrict it.  So testing for the main stack took an inordinate amount of time (a minute or two). This led to the choice of using the fibre stack for the official suite if there was going to be just one spec.

Please let me know which of these you'd like me to add.

Reply:
@damaxwell I'd go for 1, 3 and 5. Speeding up the stack overflow by allocating a 512 byte staticarray should help the speed of the test on the BSDs, and testing main and fiber stacks is a good idea.

Reply:
The new specs are added, @RX14.  It was FreeBSD that seems to have a default stack size of 0.5G, but `ulimit` works just fine to limit it; not sure what I had done wrong.   It takes 2 minutes (in a VM) to exhaust a stack that big. I left a comment in the spec to note the potential trouble and remedy.